### PR TITLE
feat(toggle directives): allow simple elements to use directive

### PR DESCRIPTION
**Pull request by mosinve**
sketch to test:

    <div>
      <p>
      <b-btn v-b-toggle.collapse1 variant="primary">Toggle Collapse</b-btn>
      </p>
      <p v-b-toggle.collapse1 >
    i can open collapse too
      </p>

      <b-collapse id="collapse1">
      <b-card>
          Collapse contents Here
      </b-card>
      </b-collapse>
    </div>

### CODE REVIEW

**Comment by mosinve** [comment]
sketch to test:

`<div>
  <p>
  <b-btn v-b-toggle.collapse1 variant="primary">Toggle Collapse</b-btn>
  </p>
  <p v-b-toggle.collapse1 >
i can open collapse too
  </p>

  <b-collapse id="collapse1">
  <b-card>
      Collapse contents Here
      <b-btn v-b-toggle.collapse2 size="sm">Toggle Inner Collapse</b-btn>
      <b-collapse id=collapse2 class="mt-2">
      <b-card>Hello!</b-card>
      </b-collapse>
  </b-card>
  </b-collapse>
</div>

`

**Comment by tmorehouse** [comment]
Looks good.

One extra change needed. `v-b-modal` also uses the `_targets.js` helper.  Will need to alter https://github.com/bootstrap-vue/bootstrap-vue/blob/master/lib/directives/modal.js#L7-L11 to use vnode as well

**Comment by mosinve** [comment]
True.

File: lib/directives/toggle.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 const listen_types = {click: true};
 
 export default {
-    bind(el, binding) {
-        const targets = target(el, binding, listen_types, ({targets, vm}) => {
+    bind(el, binding, vnode) {
+        const targets = target(vnode, binding, listen_types, ({targets, vnode}) => {
             targets.forEach(target => {
+                const vm = vnode.context;
                 vm.$root.$emit('collapse::toggle', target);
```
Comment (tmorehouse) [inline lib/directives/toggle.js]: You could also just do:
```js 
    vnode.context.$root.$emit('collapse::toggle', target);
```

And save on using a variable :)

Comment (tmorehouse) [inline lib/directives/toggle.js]: Also in the section below, you could also replace `el.__vue__` with `vnode.context` on line 17, 

and remove the test for `el.__vue__` on line 14

Comment (mosinve) [inline lib/directives/toggle.js]: Yep, you absolutely right :)

**Pull request merged**